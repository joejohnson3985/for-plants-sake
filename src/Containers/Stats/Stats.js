import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Stats.scss'


export class Stats extends Component {

  getStats = (option) => {

    let stat = this.props.trails.reduce((accum, trail) => {
      accum += trail[option]
      return accum
    }, 0)

    return stat
  }

  handleClick = (e) => {
    window.history.back()
  }



  render() {

    return(
      <div className='overlay'>
        <div className='popup'>
            <p>You have hiked <strong>{this.props.trails.length}</strong> trails, covering <strong>{this.getStats('length')}</strong> miles and climbing <strong>{this.getStats('ascent')}</strong> feet. You rock!</p>
            <button onClick={this.handleClick}>Close</button>

        </div>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  trails: state.trails
})

export default connect(mapStateToProps)(Stats)