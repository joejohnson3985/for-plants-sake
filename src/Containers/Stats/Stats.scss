@import '../../Utilities/_Variables.scss';


.overlay {
    @include popup;
    background-color: rgba(33,33,33, 0.9);
    width: 100vw;
    height: 190vh;

    .popup {
      padding: 30px 50px;
      margin-top: 50%;
      border: 2px solid $primary;
      background: $text-bg;
      color: $primary;
    }
  }