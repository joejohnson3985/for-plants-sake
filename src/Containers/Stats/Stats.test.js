import React from 'react';
import {Stats, mapDispatchToProps} from './Stats';
import { shallow } from 'enzyme';
import MockData from '../../Utilities/mockData.js'

let wrapper;
let mockTrails = MockData.mockTrails

describe('Stats', () => {
  beforeEach(() => {
    wrapper = shallow(
      <Stats trails={mockTrails} />
    )

  })
  describe('Component', () => {
    it('Should match snapshot',() => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})