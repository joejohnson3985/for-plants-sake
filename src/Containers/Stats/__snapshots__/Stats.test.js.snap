// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Component Should match snapshot 1`] = `
<div
  className="overlay"
>
  <div
    className="popup"
  >
    <p>
      You have hiked 
      <strong>
        2
      </strong>
       trails, covering 
      <strong>
        11
      </strong>
       miles and climbing 
      <strong>
        3802
      </strong>
       feet. You rock!
    </p>
    <button
      onClick={[Function]}
    >
      Close
    </button>
  </div>
</div>
`;
